apollo-normalized-cache-api/src/commonMain/kotlin/com/apollographql/apollo/cache/normalized/RecordFieldJsonAdapter.kt

```diff
@@ -0,0 +1,84 @@
+package com.apollographql.apollo.cache.normalized
+
+import com.apollographql.apollo.api.internal.Throws
+import com.apollographql.apollo.api.internal.json.BufferedSourceJsonReader
+import com.apollographql.apollo.api.internal.json.JsonWriter
+import com.apollographql.apollo.api.internal.json.use
+import com.apollographql.apollo.cache.normalized.internal.CacheJsonStreamReader
+import okio.Buffer
+import okio.BufferedSource
+import okio.ByteString.Companion.encodeUtf8
+import okio.IOException
+import kotlin.jvm.JvmStatic
+
+/**
+ * An adapter used to serialize and deserialize Record fields. Record object types will be serialized to
+ * [CacheReference].
+ */
+class RecordFieldJsonAdapter {
```

Comment:
should it be an object?

Reply:
I could potentially make it an object but it is not under `internal`. I'm afraid that would break compatibilty.